### DISCUSSION TITLE
Office365 email account: I can't log in

### DISCUSSION THREAD

**Discussion by davide-alghi** [YetiForceCompany/YetiForce#69] [How to]
Hi everyone,
I'm trying to use RoundCube with an Office365 email account, but I haven't succeeded yet.

I created a webservice-application
![Schermata del 2024-11-27 20-44-23](https://github.com/user-attachments/assets/29c196a0-c5a7-461c-b72c-a8eb70caccbd)

and then the mail server for O365.
![Schermata del 2024-11-27 20-45-06](https://github.com/user-attachments/assets/9705c4aa-5226-434e-b05f-8d13bcf50764)

but when I try to login from RoundCube it gives an error.
![image003](https://github.com/user-attachments/assets/52d21a42-3c5b-4a54-ad60-e018b1366a92)

I tried to configure the Azure account as multi-tenant,
![image016](https://github.com/user-attachments/assets/ccdf7ede-89e5-429f-b902-25ab8fe30c13)

but nothing: still an error.
![image001](https://github.com/user-attachments/assets/905d4508-a3d8-40bc-bf7d-e324969b09c7)

Can anyone give me instructions on how to configure Azure to make it talk to YetiForce?
Any suggestions are welcome.
tnx.

**Comment by MonsterITServices** [comment]
Did you put in an endpoint URL in azure?

**Reply by davide-alghi** [reply]
hi @MonsterITServices,
tnx for your reply.
Yes, of course. I got it from here

![390550081-9705c4aa-5226-434e-b05f-8d13bcf50764](https://github.com/user-attachments/assets/0480e109-9d82-4d90-8e59-43cd699a41e1)

**Comment by davide-alghi** [comment]
When I try to access via RoundCube, the system log says

`2024-12-11 16:46:27.0068 [error][] - TypeError: App\Mail\Account::getAccessToken(): Return value must be of type string, null returned in /var/www/html/yetiforce/app/Mail/Account.php:249
Stack trace:
#0 /var/www/html/yetiforce/app/Mail/Account.php(305): App\Mail\Account->getAccessToken()
#1 /var/www/html/yetiforce/modules/OSSMail/models/Record.php(172): App\Mail\Account->openImap()
#2 /var/www/html/yetiforce/modules/OSSMail/actions/CheckMails.php(34): OSSMail_Record_Model::updateMailBoxCounter()
#3 /var/www/html/yetiforce/include/main/WebUI.php(161): OSSMail_CheckMails_Action->process()
#4 /var/www/html/yetiforce/index.php(26): Vtiger_WebUI->process()
#5 /var/www/html/yetiforce/public_html/index.php(13): require('...')
#6 {main}`

**Reply by rskrzypczak** [reply]
FYI these logs do not come from roundcube, but from the functionality of displaying the number of unread messages, which should appear in the page header.
![image](https://github.com/user-attachments/assets/59a14d7b-1e2a-4347-9b6e-b15a18508974)

**Comment by rskrzypczak** [comment]
@davide-alghi you shouldn't mark anything here

![image](https://github.com/user-attachments/assets/622d9158-56a5-4b61-933b-13198242e6d6)


YetiForce not using implicit flow.

First, try creating an email account and see if you can use the buttons highlighted in the screenshot.
![image](https://github.com/user-attachments/assets/9ad4656c-17ab-4070-843e-0662165fb754)

**Comment by davide-alghi** [comment]
@rskrzypczak thanks,
I'll try your suggestion as soon as possible.

**Comment by davide-alghi** [comment]
I followed your suggestion

- from **Mail tools > Mail - Scanner** i started the authentication process by clicking on **Authorization** button
![Schermata del 2024-12-13 17-35-02](https://github.com/user-attachments/assets/e1acf310-78a2-4632-b7d5-6b53f7e74b29)
- the page redirects to **https://login.microsoftonline.com**
![Schermata del 2024-12-13 17-35-09](https://github.com/user-attachments/assets/d7234084-3ce0-455e-b3d2-4d4b7d563a4a)
- I exec login process (2FA)
- but it still returns error

there is this info in the error (???)
`(Yeti with Davide) is not configured as a multi-tenant application. Usage of the /common endpoint is not supported for such applications created after '10/15/2018'. Use a tenant-specific endpoint or configure the application to be multi-tenant.`

the error from system.log

`2024-12-13 16:41:22.0968 [error][] - No permission or wrong data
Api\Core\Exception: No permission or wrong data in api/webservice/OAuth/BaseAction/MailAccount.php:98
Stack trace:
#0 api/webservice/Core/BaseAction.php(46): Api\OAuth\BaseAction\MailAccount->checkPermission()
#1 api/webservice/Controller.php(132): Api\Core\BaseAction->checkAction()
#2 webservice.php(25): Api\Controller->process()
#3 public_html/webservice.php(13): require('...')
#4 {main}
2024-12-13 16:41:22.0875 [info][[application]] - 
$_GET = ['_container'=>'OAuth','module'=>'MailAccount','action'=>'XXXXXXXXXXXXXXXXXXXXX','error'=>'invalid_request','error_description'=>'AADSTS50194: Application \'XXXXXXXXXXXXXXXXXXX\'(Yeti with Davide) is not configured as a multi-tenant application. Usage of the /common endpoint is not supported for such applications created after \'10/15/2018\'. Use a tenant-specific endpoint or configure the application to be multi-tenant. Trace ID: XXXXXXXXXXXXXXXXX Correlation ID: XXXXXXXXXXXXXXX Timestamp: 2024-12-13 15:41:21Z','state'=>'72b74a2ad1d05b9081121fb00fe53cbdcb2f']
$_COOKIE = ['YTSID'=>'XXXXXXXXXXXXXXXXXXXXXXX']
$_SESSION = ['CSP_TOKEN'=>'XXXXXXXXXXXXXXXXXXXXX','last_activity'=>1734104436.778327]
_HEADERS = ['Host'=>'yetiforce.XXXXXXXX.com','Priority'=>'u=0, i','Cookie'=>'YTSID=XXXXXXXXXXXXXXXXX','Accept-Encoding'=>'gzip, deflate, br, zstd','Referer'=>'https://login.microsoftonline.com/','Sec-Ch-Ua-Platform'=>'"Windows"','Sec-Ch-Ua-Mobile'=>'?0','Sec-Ch-Ua'=>'"Brave";v="131", "Chromium";v="131", "Not_A Brand";v="24"','Sec-Fetch-Dest'=>'document','Sec-Fetch-Mode'=>'navigate','Sec-Fetch-Site'=>'cross-site','Accept-Language'=>'en-US,en;q=0.5','Sec-Gpc'=>'1','Accept'=>'text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8','User-Agent'=>'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/131.0.0.0 Safari/537.36','Upgrade-Insecure-Requests'=>'1','Cache-Control'=>'max-age=0','Authorization'=>'']
$stability = ['error_reporting'=>'E_ALL & ~E_NOTICE','output_buffering'=>'4096']
$security = ['session.cookie_httponly'=>'Off']
$libraries = ['pdo_sqlsrv'=>'LBL_NO','mongodb'=>'LBL_NO']
$environment = ['currentUser'=>' (uid:33)']
====================================================================================================================================


`
